extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -14,12 +14,25 @@
 using Microsoft.Azure.Functions.Worker.Core;
 using Microsoft.Extensions.Options;
 using Microsoft.Extensions.Logging;
+using Microsoft.Azure.Functions.Worker.Extensions.Abstractions;
 
 namespace Microsoft.Azure.Functions.Worker
 {
     /// <summary>
     /// Converter to bind Blob Storage type parameters.
     /// </summary>
+    [SupportsDeferredBinding]
+    [SupportedConverterTypes(
+        typeof(string), typeof(IEnumerable<string>), typeof(string[]),
```

Comment:
What happens if the user binds some parameters to supported converter types, and others to unsupported types? I'm thinking of a scenario where a user binds to `ServiceBusReceivedMessage` but also binds to a various string properties. Curious if this is a supported scenario.

Reply:
Can you share a basic function with the scenario you are mentioning to make sure I understand it correctly? I think it should bind per parameter, but I will double check.

Reply:
For the sample blob function below, Stream supports deferred binding while BinaryData does not support it. This will work.

```csharp
        [Function(nameof(ExpressionFunction))]
        public void Run(
            [BlobTrigger("expression-trigger")] Stream book,
            [BlobInput("input-container/sample1.txt")] BinaryData myBlob,
            FunctionContext context)
        {
            _logger.LogInformation("Trigger content: {content}", book);
            _logger.LogInformation("Blob content: {content}", myBlob);
        }
```

```json
   "bindings": [
      {
        "name": "book",
        "direction": "In",
        "type": "blobTrigger",
        "path": "expression-trigger",
        "properties": {
          "supportsDeferredBinding": "True"
        }
      },
      {
        "name": "myBlob",
        "direction": "In",
        "type": "blob",
        "blobPath": "input-container/sample1.txt",
        "cardinality": "One",
        "properties": {}
      }
    ]
 
```

Reply:
Sounds good - in the example I was thinking of there would only be one attribute, but multiple parameters, e.g.:
```c#
public void Run(
            [ServiceBusTrigger("expression-trigger")] ServiceBusReceivedMessage message,
            string MessageId,
            string LockToken,
            FunctionContext context)
```

Reply:
It's a good scenario. For Blob, I have a similar function handy. In case of below function, `Stream` is the binding type and `name` is part of the expression. Irrespective of the type of `name`, if Stream supports deferred binding, it will get a `true` in metadata, otherwise `false`.

```csharp
        [Function(nameof(BlobStreamFunction))]
        public async Task BlobStreamFunction([BlobTrigger("stream-trigger/{name}")] Stream stream, string name)
```

I am interested to know if we can infer similarly for ServiceBus or not. Meaning `message` will be considered as Binding type and `messageId`, `lockToken` are part of expression? Let me know if this not as you expected. We can definitely discuss this offline as well. Thanks for bringing this up.

Reply:
The other parameters are not part of an expression. Instead they are intended to bind to properties of the message. I think this scenario is probably unusual because users can get the same properties from the ServiceBusReceivedMessage type itself. I mostly was just curious as to what would happen in this case - I think it is reasonable if we say it is not supported.

Reply:
Had a quick chat with Fabio on this. There will be an upcoming change for custom binding sources where such scenarios will be supported. Binding source will help identifying where is the parameter coming from. For example, is it coming from body/query/header and so on. Will share if I find any docs related to it.